batch_job/src/batch_job_amazon.c

```diff
@@ -92,13 +108,115 @@ static const char * aws_instance_select( int cores, int memory, int disk )
 
 	for(i=aws_instance_table;i->name;i++) {
 		if(cores<=i->cores && memory<=i->memory) {
-			debug(D_BATCH,"job requiring CORES=%d MEMORY=%d matches instance type %s",cores,memory,i->name);
+			debug(D_BATCH,"job requiring CORES=%d MEMORY=%d matches instance type %s\n",cores,memory,i->name);
 			return i->name;
 		}
 	}
 	return 0;
 }
 
+/*
+Lookup aws_instance_type given instance type name
+*/
+
+static struct aws_instance_type * aws_instance_lookup(const char *instance_type)
+{
+	struct aws_instance_type *i;
+	for(i=aws_instance_table;i->name;i++) {
+		if(strcmp(instance_type, i->name) == 0) {
+			return i;
+		}
+	}
+	debug(D_BATCH,"instance type %s not found in instance type table\n", instance_type);
+	return 0;
+}
+
+/*
+Compare if instance of type 1 does fit in instance of type 2.
+On fitting (cores and memory of type 1 instance <= type 2)
+return 1, otherwise return null
+*/
+
+static int compare_instance_type(struct aws_instance* instance, const char* type_2){
+	const char *type_1 = strdup(instance->instance_type);
```

Comment:
You don't need the strdup here. In fact it causes a memory leak as it is never freed.